### PR TITLE
Redone dev setup.

### PR DESCRIPTION
**Pull request by ronilan**
This pull request redones the Mac dev environment.

1. Everything related to dev is in dev directory.
  - Dev container has build tools, docker, docker compose and multiple versions of node installed.
 - There is simple way to get local "one off" containers.
 - There is an easy way to work with "formalized" dev reop.

4. Four npm commands launch four shell scripts (e53e65d):
a. `npm run dev` - starts and ubuntu based dev environment defined in a docker compose. Code is mounted. This is where work should be done.
b. `npm run dev:oneoff` - starts a "one off" dev environment. code is copied. This is for testing specific node/os things.
c. `npm run dev:repo:reset` - runs a set of git operations that "resets" the dev repo to match the main repo.
d. `npm run dev:reinstall` - removed node_modules and runs npm install. To be used after switching node versions.

The setup is similar to the one used in downstream repo https://github.com/appoptics/appoptics-apm-node/pull/145

### CODE REVIEW

File: dev/repo/reset.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
```
Comment (prettycoder) [inline dev/repo/reset.sh:1]: I think I remember why we have this in this repo, it is for testing out github actions

**Review by prettycoder — APPROVED** [review]
lgtm

some comments as in other PR

File: dev/repo/reset.sh (hunk 1 above)
Comment (ronilan) [reply dev/repo/reset.sh:1]: It was indeed originally for that and now formalized to do a little more (see [WIP docs](https://github.com/appoptics/appoptics-bindings-node/tree/AO-20192-Update-Documentation-no-build#dev-repo)). 

Given that the repo is the *"production environment"* for GitHub Actions, makes sense to have *"development environment"* somewhere. I tried [Act](https://github.com/nektos/act) to run things locally, but it is lacking, so for now I guess dev repos ae the way to go for complex GitHub Action flows...

**Review by prettycoder — APPROVED** [review]
lgtm

**Pull request merged**